src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationServiceRunner.cs

```diff
@@ -27,6 +30,117 @@ internal abstract class AbstractRecommendationServiceRunner<TSyntaxContext>
 
         public abstract ImmutableArray<ISymbol> GetSymbols();
 
+        protected abstract bool IsLambdaExpression(SyntaxNode node);
+
+        protected abstract bool IsInvocationExpression(SyntaxNode node);
+
+        protected abstract bool TryGetOrdinalInArgumentList(SyntaxNode argumentOpt, out int ordinalInInvocation);
+
+        // This code is to help give intellisense in the following case: 
+        // query.Include(a => a.SomeProerty).ThenInclude(a => a.
+        // where there are more than one overloads of ThenInclude accepting different types of parameters.
+        protected ImmutableArray<ISymbol> GetSymbols(IParameterSymbol parameter, int position)
+        {
+            // Starting from a. in the example, looking for a => a.
+            if (!(parameter.ContainingSymbol is IMethodSymbol containingMethod &&
+                containingMethod.MethodKind == MethodKind.AnonymousFunction))
+            {
+                return default;
+            }
+
+            // Cannot proceed without DeclaringSyntaxReferences.
+            // We expect that there is a single DeclaringSyntaxReferences in the scenario.
+            // If anything changes on the compiler side, the approach should be revised.
+            if (containingMethod.DeclaringSyntaxReferences.IsDefaultOrEmpty || containingMethod.DeclaringSyntaxReferences.Length > 1)
```

Comment:
DeclaringSyntaxReferences is never Default.  So you could just simple this to `.DeclaringSyntaxReferences.Length != 1`